test/datasets/test_web_qsp_dataset.py

```diff
@@ -0,0 +1,201 @@
+        "answer": train_answers
+    }
+    val_graphs = {
+        "graph": val_triplets,
+        "question": val_questions,
+        "answer": val_answers
+    }
+    test_graphs = {
+        "graph": test_triplets,
+        "question": test_questions,
+        "answer": test_answers
+    }
+
+    from datasets import Dataset, DatasetDict, load_from_disk
+
+    ds_train = Dataset.from_dict(train_graphs, split="train")
+    ds_val = Dataset.from_dict(val_graphs, split="validation")
+    ds_test = Dataset.from_dict(test_graphs, split="test")
+
+    ds = DatasetDict({
+        "train": ds_train,
+        "validation": ds_val,
+        "test": ds_test
+    })
+
+    def mock_load_dataset(name: str):
+        # Save the dataset and then load it to emulate downloading from HF
+        DATASET_CACHE_DIR = os.path.join(tmp_path,
+                                         ".cache/huggingface/datasets", name)
+        os.makedirs(DATASET_CACHE_DIR, exist_ok=True)
+
+        ds.save_to_disk(DATASET_CACHE_DIR)
+        dataset_remote = load_from_disk(DATASET_CACHE_DIR)
+        return dataset_remote
+
+    return mock_load_dataset, ds
+
+
+@onlyRAG
+@withPackage("datasets", "pandas")
+def test_kgqa_base_dataset(tmp_path, monkeypatch):
+
+    num_nodes = 500
+    num_edge_types = 25
+    num_trips = 5000
+
+    # Mock the dataset graphs
+    mock_load_dataset_func, expected_result = create_mock_graphs(
+        tmp_path, train_size=10, val_size=5, test_size=5, num_nodes=num_nodes,
+        num_edge_types=num_edge_types, num_trips=num_trips)
+
+    import datasets
+
+    monkeypatch.setattr(datasets, "load_dataset", mock_load_dataset_func)
+
+    # Mock the SentenceTransformer
+    import torch_geometric.datasets.web_qsp_dataset
+    monkeypatch.setattr(torch_geometric.datasets.web_qsp_dataset,
+                        "SentenceTransformer", MockSentenceTransformer)
+
+    dataset_train = KGQABaseDataset(root=tmp_path, dataset_name="TestDataset",
+                                    split="train", use_pcst=False)
+    assert len(dataset_train) == 10
+    assert str(dataset_train) == "KGQABaseDataset(10)"
+    for graph in dataset_train:
+        assert graph.x.shape == (num_nodes, 1024)
+        assert graph.edge_index.shape == (2, num_trips)
+        assert graph.edge_attr.shape == (
+            num_trips, 1024)  # Reminder: edge_attr encodes the entire triplet
+
+    dataset_val = KGQABaseDataset(root=tmp_path, dataset_name="TestDataset",
+                                  split="val", use_pcst=False)
+    assert len(dataset_val) == 5
+    assert str(dataset_val) == "KGQABaseDataset(5)"
+
+    dataset_test = KGQABaseDataset(root=tmp_path, dataset_name="TestDataset",
+                                   split="test", use_pcst=False)
+    assert len(dataset_test) == 5
+    assert str(dataset_test) == "KGQABaseDataset(5)"
+
+    # TODO: More rigorous tests to validate that values are correct
+    # TODO: Proper tests for PCST and CWQ
```

Comment:
```suggestion
    # TODO(zaristei): More rigorous tests to validate that values are correct
    # TODO(zaristei): Proper tests for PCST and CWQ
```
https://google.github.io/styleguide/pyguide.html#312-todo-comments